modules/oe_media_embed/src/Form/MediaEmbedDialog.php

```diff
@@ -336,18 +348,45 @@ public function buildEmbedStep(array $form, FormStateInterface $form_state): arr
     // @todo add element alignment and caption if the relevant filters are
     // enabled. See EntityEmbedDialog for example.
     // Allow to specify a view mode if the media type has more than 1.
-    $options = $this->getMediaViewModeOptions($entity);
-    if ($options) {
-      $form['attributes']['data-entity-view-mode'] = [
-        '#type' => 'select',
-        '#title' => $this->t('Display as'),
-        '#options' => $options,
-        '#default_value' => $entity_element['data-entity-view-mode'],
-        '#required' => TRUE,
-        '#access' => TRUE,
+    $display_options = $this->getMediaViewModeOptions($entity);
+
+    // If no view mode was found for this media type,
+    // prompt the user to enable one.
+    if (empty($display_options)) {
+      $form['attributes']['data-entity-view-mode-warning'] = [
+        '#markup' => '<div>' . t('There is no embedable view mode for this media type.') . '</div>',
       ];
+      if ($this->moduleHandler->moduleExists('field_ui')) {
+        $form['attributes']['data-entity-view-mode-link'] = [
+          '#type' => 'link',
+          '#title' => t('Manage @media view modes', ['@media' => $entity->bundle()]),
+          '#url' => Url::fromRoute('entity.entity_view_display.' . $entity->getEntityTypeId() . '.default', [
+            'media_type' => $entity->bundle(),
+          ]),
+          '#attributes' => ['target' => '_blank'],
+        ];;
+      }
+    }
+    else {
```

Comment:
Could we use here like 
```suggestion
    elseif(count($display_options) === 1) {
```
for simplicity?